src/object/schema_manager.c

```diff
@@ -5861,20 +5867,21 @@ sm_find_index (MOP classop, char **att_names, int num_atts, bool unique_index_on
 	    }
 
 	  att2 = classobj_find_attribute (class_, att_names[i], 0);
-	  if (att2 == NULL || att1->id != att2->id)
+	  if (att2 && att1->id == att2->id)
 	    {
+	      found = true;
 	      break;
 	    }
 	}
 
-      if ((i == num_atts) && con->attributes[i] == NULL)
+      if (found)
```

Comment:
how about?
if ((i == num_atts) && con->attributes[i] == NULL)
==> if (i == num_atts)

Reply:
it looks better to write the logic "if (i == 0 || i == num_atts)" instead of using "found" variable.